servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -4949,6 +5019,17 @@ void validatePolicyAssertion(Assertion assertion, String caller) {
             throw ZMSUtils.requestError("Assertion action contains control characters: "
                     + resource, caller);
         }
+
+        // we are not going to allow any user to add/update
+        // an assertion that the associated role does not exist
+        int endDomainName = assertion.getRole().indexOf(":");
+        String domain = assertion.getRole().substring(0, endDomainName);
```

Comment:
1) we need to check here for -1. if the user just passes "role1" for the role name instead of "athenz:role.role1" then we'll get index out of bounds exception.

2) also please add a unit test case for that scenario.